pkg/runtime/depot.go

```diff
@@ -422,3 +465,39 @@ func someFilesExist(filePaths []string, basePath string) bool {
 	}
 	return false
 }
+
+// removeStaleArtifacts iterates over all unused artifacts in the depot, sorts them by last access
+// time, and removes them until the size of cached artifacts is under the limit.
+func (d *depot) removeStaleArtifacts() {
+	type artifact struct {
+		id   strfmt.UUID
+		info *artifactInfo
+	}
```

Comment:
Suggest encoding id in artifactInfo and avoiding this type altogether, though I might be missing something here..

Reply:
The depot cache is a map of artifact IDs to artifactInfo. I didn't want to duplicate information, but I can add it as a non-json-tagged field so it's not duplicated in the cache on disk at least.